### DISCUSSION TITLE
Trying and failing to compile for a converter

### DISCUSSION THREAD

**Discussion by bogorad** [manna-harbour/miryoku#369] [Q&A] [answered]
I'm trying to compile Miryoku-qmk for [adafruit_rp2040_usbh](https://github.com/whyaaronbailey/adafruit_rp2040_usbh) converter. Can someone please explain the steps? I'm getting errors.

I tried installing a clean QMK, the firmware version is 0.27.1 (adafruit_rp2040_usbh suggests at least v0.24), following the  setup steps, then copying the `manna-harbour_miryoku` files into `~/qmk_firmware/users/manna-harbour_miryoku`, and `keyboards/converter/adafruit_rp2040_usbh/keymaps/manna-harbour_miryoku`

` qmk compile -c -kb converter/adafruit_rp2040_usbh -km default` works,

but

` qmk compile -c -j6 -kb converter/adafruit_rp2040_usbh -km manna-harbour_miryoku` fails:

https://pastebin.com/bScAa0Ap

specifically,

```
Compiling: quantum/keymap_introspection.c                                                          
In file included from <command-line>:
./keyboards/converter/adafruit_rp2040_usbh/keymaps/manna-harbour_miryoku/config.h:
14:1: error: implicit declaration of function 'LAYOUT_split_3x5_3' [-Werror=implicit-function-declaration]
   14 | LAYOUT_split_3x5_3(\
      | ^~~~~~~~~~~~~~~~~~
users/manna-harbour_miryoku/manna-harbour_miryoku.h:
17:37: note: in expansion of macro 'LAYOUT_miryoku'
   17 | #define U_MACRO_VA_ARGS(macro, ...) macro(__VA_ARGS__)
      |                                     ^~~~~

```

What is the proper way of combining the two?

upd: same result when using https://github.com/drashna/qmk_userspace/tree/miryoku

**Comment by manna-harbour** [comment, marked as answer]
What did you put in `keyboards/converter/adafruit_rp2040_usbh/keymaps/manna-harbour_miryoku`? You need a mapping there that corresponds to how you want to use the keyboard. What keyboard are you going to use it with, and how do you want to use Miryoku with it? E.g. https://github.com/manna-harbour/miryoku_qmk/tree/miryoku/users/manna-harbour_miryoku#60_ansi. Also see https://github.com/manna-harbour/xmk.
